File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/aas/metamodel/hashmap/aas/qualifier/AdministrativeInformation.java
package org.eclipse.basyx.aas.metamodel.hashmap.aas.qualifier;

import java.util.HashMap;
import java.util.HashSet;

import org.eclipse.basyx.aas.api.metamodel.aas.qualifier.IAdministrativeInformation;
import org.eclipse.basyx.aas.api.metamodel.aas.reference.IReference;
import org.eclipse.basyx.aas.metamodel.facades.AdministrativeInformationFacade;
import org.eclipse.basyx.aas.metamodel.facades.HasDataSpecificationFacade;

/**
 * AdministrativeInformation class
 * 
 * @author kuhn
 *
 */
public class AdministrativeInformation extends HashMap<String, Object> implements IAdministrativeInformation {

	/**
	 * Version of serialized instances
	 */
	private static final long serialVersionUID = 1L;
	
	public static final String VERSION="version";
	
	public static final String REVISION="revision";

	/**
	 * Constructor
	 */
	public AdministrativeInformation() {
		// Add qualifier
		putAll(new HasDataSpecification());

		// Default values
		put(VERSION, "");
		put(REVISION, "");
	}

	/**
	 * Constructor
	 */
	public AdministrativeInformation(String version, String revision) {
		// Add qualifier
		putAll(new HasDataSpecification());

		// Default values
		put(VERSION, version);
		put(REVISION, revision);
	}

	@Override
	public HashSet<IReference> getDataSpecificationReferences() {
		return new HasDataSpecificationFacade(this).getDataSpecificationReferences();
	}

	@Override
	public void setDataSpecificationReferences(HashSet<IReference> ref) {
		new HasDataSpecificationFacade(this).setDataSpecificationReferences(ref);
		
	}

	@Override
	public void setVersion(String version) {
		new AdministrativeInformationFacade(this).setVersion(version);
		
	}

	@Override
	public String getVersion() {
	return new AdministrativeInformationFacade(this).getVersion();
	}

	@Override
	public void setRevision(String revision) {
		new AdministrativeInformationFacade(this).setRevision(revision);
		
	}

	@Override
	public String getRevision() {
		return new AdministrativeInformationFacade(this).getRevision();
	}


}

File: sdks/c++/basys.sdk.cc/src/aas/impl/metamodel/hashmap/aas/qualifier/AdministrativeInformation.cpp
/*
 * AdministrativeInformation.cpp
 *
 *      Author: wendel
 */

#include "AdministrativeInformation.h"

namespace basyx {
namespace aas {
namespace qualifier {
namespace impl {

AdministrativeInformation::AdministrativeInformation() :
  data_specification_references()
{}

AdministrativeInformation::AdministrativeInformation(const std::string & version, const std::string & revision) :
  data_specification_references(),
  version {version},
  revision {revision}
{}

AdministrativeInformation::AdministrativeInformation(basyx::object::object_map_t & map)
{
  this->revision = map.at(qualifier::internalAdministrationPaths::REVISION).GetStringContent();
  this->version = map.at(qualifier::internalAdministrationPaths::VERSION).GetStringContent();
  this->data_specification_references = map.at(HasDataSpecificationPaths::HASDATASPECIFICATION).Get<basyx::specificCollection_t<reference::IReference>>();
}

void AdministrativeInformation::setDataSpecificationReferences(const basyx::specificCollection_t<reference::IReference>& data_specification_references)
{
  this->data_specification_references = data_specification_references;
}

void AdministrativeInformation::setVersion(const std::string & version)
{
  this->version = version;
}

void AdministrativeInformation::setRevision(const std::string & revision)
{
  this->revision = revision;
}

basyx::specificCollection_t<reference::IReference> AdministrativeInformation::getDataSpecificationReferences() const
{
  return this->data_specification_references;
}

std::string AdministrativeInformation::getVersion() const
{
  return this->version;
}

std::string AdministrativeInformation::getRevision() const
{
  return this->revision;
}

}
}
}
}